pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,427 @@
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_DCI = Instrument.measurement(
+        MODES["DCI"],
+        """Return the measured value for DC current""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_DCV = Instrument.measurement(
+        MODES["DCV"],
+        """Return the measured value for DC Voltage""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_R2W = Instrument.measurement(
+        MODES["R2W"],
+        """Return the measured value for 2-wire resistance""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_R4W = Instrument.measurement(
+        MODES["R4W"],
+        """Return the measured value for 4-wire resistance""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_Rext = Instrument.measurement(
+        MODES["Rext"],
+        """Return the measured value for extended resistance mode (>30M, 2-wire) resistance""",
+        get_process = (lambda x: float(x)),
+        )
+
+    mode = Instrument.control(
+        "B",
+        "%s",
+        """Return current selected measurment mode, this propery can be set""",
+        get_process = (lambda x: get_mode(x.encode(encoding="ASCII"))),
+        set_process = (lambda x: MODES[x]),
+        validator = strict_discrete_set,
+        values = MODES,
+        )
+
+    #TODO: fix validators
+    range = Instrument.control(
+        "B",
+        "%s",
+        """Returns the current measurment range, this property can be set""",
+        get_process = (lambda x: get_range(x.encode(encoding="ASCII"))),
+        set_process = (lambda x,y: RANGES[x][y]),
+        # validator = joined_validators(strict_discrete_set,strict_discrete_set),
+        # values = RANGES,
+        )
+
+    resolution = Instrument.control(
+        "B",
+        "N%d",
+        """Return current selected resolution, this property can be set""",
+        get_process = (lambda x: 6-get_status(x.encode(encoding="ASCII"),"digits")),
+        validator = strict_discrete_set,
+        values = [3,4,5],
+        )
+
+    status = Instrument.measurement(
+        "B",
+        """Checks the status registers""",
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"))),
+        )
+
+    SRQ_mask = Instrument.control(
+        "B",
+        "M%o",
+        """Return current SRQ mask, this property can be set""",
+        get_process = (lambda x: get_status(x.encode(encoding="ASCII"),"SRQ")),
+        validator = strict_range,
+        values = [0,63],
+        )
+
+    trigger = Instrument.control(
+        "B",
+        "%s",
+        """Return current selected trigger mode, this property can be set""",
+        get_process = (lambda x: get_trigger(x.encode(encoding="ASCII"))),
+        set_process = (lambda x: TRIGGERS[x]),
+        validator = strict_discrete_set,
+        values = TRIGGERS,
+        )
+
+    #Functions using low-level access via instrument.adapter.conneciton methods
```

Comment:
typo in `connection`

Reply:
Thank you, will be fixed with the next submit